OrbitGl/TrackManager.h

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#pragma once
+
+#include <stdint.h>  // for int32_t, uint64_t, uint32_t
+#include <stdlib.h>  // for abs
+
+#include <map>     // for map
+#include <memory>  // for shared_ptr
+#include <mutex>   // for recursive_mutex
+#include <string>  // for string
+#include <unordered_map>
+#include <vector>  // for vector
+
+#include "AsyncTrack.h"
+#include "FrameTrack.h"
+#include "GpuTrack.h"
+#include "GraphTrack.h"
+#include "PickingManager.h"  // for PickingMode
+#include "SchedulerTrack.h"
+#include "StringManager.h"  // for StringManager
+#include "ThreadTrack.h"
+#include "Timer.h"            // for Timer
+#include "Track.h"            // for Track
+#include "capture_data.pb.h"  // for FunctionInfo
+
+class TimeGraph;
+
+class TrackManager {
+ public:
+  explicit TrackManager(TimeGraph* time_graph);
+
+  void Clear();
+
+  [[nodiscard]] std::vector<Track*> GetTracks() const;
+  [[nodiscard]] std::vector<Track*> GetFilteredTracks() const { return sorted_filtered_tracks_; }
+  [[nodiscard]] std::vector<ThreadTrack*> GetThreadTracks() const;
+  [[nodiscard]] std::vector<FrameTrack*> GetFrameTracks() const;
+
+  [[nodiscard]] ThreadTrack* GetTracepointsSystemWideTrack() const {
+    return tracepoints_system_wide_track_;
+  }
+
+  void SetStringManager(std::shared_ptr<StringManager> str_manager);
+  [[nodiscard]] StringManager* GetStringManager() const { return string_manager_.get(); }
+
+  void SortTracks();
+  void SetFilter(const std::string& filter);
+  void UpdateFilteredTrackList();
+
+  void UpdateTracks(uint64_t min_tick, uint64_t max_tick, PickingMode picking_mode);
+  float GetTracksHeight() const { return std::abs(tracks_height_); }
+
+  SchedulerTrack* GetOrCreateSchedulerTrack();
+  ThreadTrack* GetOrCreateThreadTrack(int32_t tid);
+  GpuTrack* GetOrCreateGpuTrack(uint64_t timeline_hash);
+  GraphTrack* GetOrCreateGraphTrack(const std::string& name);
+  AsyncTrack* GetOrCreateAsyncTrack(const std::string& name);
+  FrameTrack* GetOrCreateFrameTrack(const orbit_client_protos::FunctionInfo& function);
+
+  void AddTrack(std::shared_ptr<Track> track);
+  void RemoveFrameTrack(uint64_t function_address);
+
+  void UpdateMovingTrackSorting();
+  int FindMovingTrackIndex();
+  [[nodiscard]] std::vector<int32_t> GetSortedThreadIds();
+
+ protected:
```

Comment:
As you don't use protected fields, you can just remove that section.

Reply:
Sure!